liblumen_alloc/src/erts/term/map.rs

```diff
@@ -81,10 +81,22 @@ impl Map {
     }
 
     pub fn update(&self, key: Term, value: Term) -> Option<HashMap<Term, Term>> {
-        let mut map = self.value.clone();
-        match map.insert(key, value) {
-            Some(_) => Some(map),
-            None => None,
+        if self.is_key(key) {
```

Comment:
I thought I had included this in the other PR. Woops! I was also thinking I could re-use this for put/3, but turns out not, because I need to check the value as well there.